core/src/main/java/com/google/cloud/sql/core/CoreSocketFactory.java

```diff
@@ -152,6 +152,42 @@ public static synchronized CoreSocketFactory getInstance() {
     return coreSocketFactory;
   }
 
+  // Overloaded getInstance method to accept Credential File Location
+  public static synchronized CoreSocketFactory getInstance(String credsJsonFileLoc) {
+    if (coreSocketFactory == null) {
+      logger.info("First Cloud SQL connection, generating RSA key pair using creds file.");
+      String userCredentialFactoryClassName = System.getProperty(
+              CredentialFactory.CREDENTIAL_FACTORY_PROPERTY);
+
+      CredentialFactory credentialFactory;
+      if (userCredentialFactoryClassName != null) {
+        try {
+          credentialFactory =
+                  (CredentialFactory)
+                          Class.forName(userCredentialFactoryClassName).newInstance();
+        } catch (Exception err) {
+          throw new RuntimeException(err);
+        }
+      } else {
+        credentialFactory = new ApplicationDefaultCredentialFactory();
+      }
+      credentialFactory.getFileLoc(credsJsonFileLoc);
+      HttpRequestInitializer credential = credentialFactory.create();
+      SQLAdmin adminApi = createAdminApiClient(credential);
+      ListeningScheduledExecutorService executor = getDefaultExecutor();
+
+      coreSocketFactory =
+              new CoreSocketFactory(
+                      executor.submit(CoreSocketFactory::generateRsaKeyPair),
+                      adminApi,
+                      credentialFactory,
+                      DEFAULT_SERVER_PROXY_PORT,
+                      executor);
```

Comment:
Hmm, so this is maybe problematic because it means that every instance uses the credentials specified, even if it's not the the JDBC url. Is that what you wanted? 

If so, why is this preferable over the environment variable?

Reply:
Had come across this use-case: if there are multiple JVMs running on the same server, and if each of them would want to use different SAs to connect to different or same cloud sql instance(s), we'll not be able to use the env var.

Reply:
In this current implementation, this is still the same: each is using the same socket-factory and thus the same credentials as whatever initialized it. 

The general use-case is a bit of an anti-pattern, as each app should have a bit more isolation - if they are all using different credentials but still have access to each other credentials there isn't much gained.